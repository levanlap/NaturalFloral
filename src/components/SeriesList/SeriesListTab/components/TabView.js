import React from 'react'
import { StyleSheet, View } from 'react-native'
import { addNavigationHelpers } from 'react-navigation'
import TabBar from './TabBar'

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  tabScreen: {
    flex: 1
  }
})

export default function TabView(props) {
  const {
    game, screenProps, router, navigation
  } = props
  const { routes, index } = navigation.state
  const ActiveScreen = router.getComponentForRouteName(routes[index].routeName)

  return (
    <View style={styles.container}>
      <TabBar navigation={navigation} game={game} />
      <View style={styles.tabScreen}>
        <ActiveScreen
          {...screenProps}
          navigation={addNavigationHelpers({
            dispatch: navigation.dispatch,
            state: routes[index]
          })}
        />
      </View>
    </View>
  )
}
