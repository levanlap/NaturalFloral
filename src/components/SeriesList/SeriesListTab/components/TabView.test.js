import React from 'react'
import { View } from 'react-native'

import TabView from './TabView'

import renderer from 'react-test-renderer'

const mockedRouter = { getComponentForRouteName: () => () => (<View />) }
const mockedNavigation = { state: { routes: [{ routeName: '' }], index: 0 } }

it('renders without crashing', () => {
  const rendered = renderer.create(<TabView router={mockedRouter} navigation={mockedNavigation} />).toJSON()
  expect(rendered).toBeTruthy()
})

it('render correctly', async () => {
  const tree = renderer.create(<TabView router={mockedRouter} navigation={mockedNavigation} />).toJSON()
  expect(tree).toMatchSnapshot()
})
