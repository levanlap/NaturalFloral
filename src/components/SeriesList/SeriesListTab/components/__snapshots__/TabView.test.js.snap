// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 28,
        "margin": 10,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#388E3C",
          "borderBottomLeftRadius": 4,
          "borderColor": "#388E3C",
          "borderTopLeftRadius": 4,
          "borderWidth": 1,
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#388E3C",
              "fontFamily": "Roboto_medium",
              "fontSize": 12,
            },
            Object {
              "color": "#FFFFFF",
            },
          ]
        }
        uppercase={false}
        virtual={undefined}
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View />
  </View>
</View>
`;
